### ISSUE TITLE
Extend GCalendar / GNode

### ISSUE THREAD

**Issue by danielepolencic** [uasabi/noderosso#44]
_TL;DR: Add the ability to upsert events._

## Context

The GCalendar/GNode can create new events with the `CREATE_EVENT.V1` action. If two identical messages are sent to the node, two invites are created.
You should change the logic for the `CREATE_EVENT.V1` action so that:

- [ ] it creates the event if it doesn't exist.
- [ ] it updates the event if it exists.

## Implementation

You will have to create a new action: `CREATE_EVENT.V2`.
The action is an extension of `CREATE_EVENT.V1` with the following properties:

```ts
  upsert: Message.extend({
    topic: z.literal('CREATE_EVENT.V1'),
    payload: z.object({
      id: z.string().noempty(),
      calendarId: z.string().nonempty(),
      summary: z.string().nonempty(),
      startingAt: z.string().nonempty(),
      endingAt: z.string().nonempty().optional(),
      timezone: z.string().nonempty().optional(),
    }),
  }),
```

The `id` is the only new field.
The id should be stored in the invite as a [private extended property](https://developers.google.com/calendar/api/guides/extended-properties). When an action is received, you should [search for the id](https://developers.google.com/calendar/api/guides/extended-properties) and if any event with that id exists in the calendar. If it does, you should update it.
